src/azul/chalice.py

```diff
@@ -64,6 +79,25 @@ def test_route(self, *args, **kwargs):
         """
         return self.route(*args, enabled=self.unit_test, **kwargs)
 
+    def _register_spec(self,
+                       path: str,
+                       path_spec: Optional[JSON],
+                       method_spec: Optional[JSON],
+                       methods: Iterable[str]):
+        assert path not in self.path_specs, 'Only specify path_spec once per route path'
+
+        if path_spec is not None:
+            self.path_specs[path] = path_spec
+
+        if method_spec is not None:
+            # No need to worry about duplicate method_specs since Chalice will
+            # complain in that case
```

Comment:
That's an assumption. There should be an explicit assertion preventing dupes, without relying on this assumption that Chalice does that for us.